ddtrace/span.py

```diff
@@ -97,8 +94,12 @@ def __init__(
         self.duration_ns = None
 
         # tracing
-        self.trace_id = trace_id or _new_id()
-        self.span_id = span_id or _new_id()
+        # DEV: explicitly invoke ``__next__`` on generator vs using next(rng) as there is an overhead
```

Comment:
If this is the case, then just not treat as an iterator?

Do we gain anything from this being an iterator?

If not then I'd consider just:

```
class xorshift64s:
    def __init__(self):
        self.x = seed_value

    def rand64(self):
        self.x = shifty_stuff
        return self.x
```

I'd even hid the fact that you need to get the generator and then call "next" on it, e.g.:

```
self.trace_id = trace_id or gen_new_id()
self.span_id = span_id or gen_new_id()
```

the overhead of hitting `getattr(threading.local(), "rand64bits")` probably won't be too bad of a hit, can afford to build a nicer interface.

Reply:
also, if we wrap in a nicer interface, then it means we don't need to access a property (or generate a new class instance) when `trace_id` and `span_id` are both provided.

Reply:
This comment is out of date now.